docs/contrib/dev_environment.md

```diff
@@ -83,45 +100,80 @@ popd
 make load-data
 ```
 
-This should complete in ~1-2 minutes.
+This operation should complete in ~1-2 minutes.
 
-## Run an analysis of a sample app (example for MacOS)
+-->
 
-> [!NOTE]
->
-> We have checked in analysis runs for all sample applications so you do NOT
-> need to run analysis yourself. The instructions below are ONLY if you want to
-> recreate, this is NOT required
+## Debugging from VSCode
 
-Ensure you have the source code for the sample applications checked out locally:
+You may want to use VSCode's built-in debugger at some point (i.e. to set
+breakpoints, watch expressions, etc...). This requires some setup to get
+working.
+
+Add the following to your `launch.json`'s `"configurations"` list. (For more
+information, click [here](https://go.microsoft.com/fwlink/?linkid=830387).)
+
+```json
+{
+  "name": "Python Debugger: Attach using Process Id",
+  "type": "debugpy",
+  "request": "attach",
+  "processId": "${command:pickProcess}",
+  "justMyCode": false,
+},
+```
+
+Now, if you got the process ID of the Kai binary and used the debugger as-is,
+you would get some bizarre behavior. This is because we compile our Python
+code to a binary using [Pyinstaller](https://pyinstaller.org/en/stable/), which
+the debugger doesn't know how to handle. Additionally, the IDE simply calls the
+binary outright, with no arguments. In other words, we can't tell the IDE to
+execute `python main.py`, we can only tell it to execute `./main.py`.
+
+Thus, we need to have the IDE spawn a Python process (not a compiled binary) by
+calling the `main.py` file itself (so the IDE spawns it) to allow the debugger to
+work properly.
+
+First, make sure `kai/rpc_server/main.py` is executable with:
 
 ```sh
-cd ./samples
-./fetch_apps.py
+chmod +x kai/rpc_server/main.py
 ```
 
-This will check out the sample app source code to: `./kai_solution_server/samples/sample_repos`.
+Next, make sure you are inside your virtual environment and find the location of
+your Python interpreter:
+
+```sh
+which python
+```
 
-Next, run the analysis:
+The `python` binary should be inside your virtual environment folder.
 
 ```sh
-cd macos
-# Sets up the podman VM on MacOS so it will mount the host filesystem into the VM
-./restart_podman_machine.sh
-# Fetches kantra (our analyzer tool) and stores it in ../bin
-./get_latest_kantra_cli.sh
-cd ..
-# Analyzes all sample apps we know about, in both the 'initial' and 'solved'
-# states, expect this to run for ~2-3 hours.
-./analyze_apps.py
+/home/jonah/Projects/github.com/konveyor-ecosystem/kai-jonah/venv/bin/python
+```
+
+Next, add the following to the top of `kai/rpc_server/main.py`:
+
+```python
+#!<result of `which python`>
+```
+
+Now executing `./kai/rpc_server/main.py` will call the script directly using the
+python interpreter found in your virtual environment.
+
+Next, modify `"konveyor.kaiRpcServerPath"` inside the IDE's `settings.json` to
```

Comment:
I did not realize we could do this, actually so useful.

Reply:
that's how I have been messing things up. lol.